packages/playwright/src/errorContext.ts

```diff
@@ -92,30 +92,35 @@ export async function attachErrorContext(testInfo: TestInfoImpl, format: 'markdo
     const parsedError = error.stack ? parseErrorStack(error.stack, path.sep) : undefined;
     const inlineMessage = stripAnsiEscapes(parsedError?.message || error.message || '').split('\n')[0];
     const location = parsedError?.location || { file: testInfo.file, line: testInfo.line, column: testInfo.column };
-    const source = await loadSource(location.file, sourceCache);
-    const codeFrame = codeFrameColumns(
-        source,
-        {
-          start: {
-            line: location.line,
-            column: location.column
+    let source = await loadSource(location.file, sourceCache);
+    // If the error location is not available on the disk (e.g. fake page.evaluate in-browser error), then fallback to the test file.
+    if (!source)
+      source = await loadSource(testInfo.file, sourceCache);
+    if (source) {
+      const codeFrame = codeFrameColumns(
+          source,
+          {
+            start: {
+              line: location.line,
```

Comment:
I think we should synchronize `location` and `source` variables - try loading source from the `parsedError.location` first, and then fallback to the test location and source.